### PR TITLE
Add base representation and update dataframe-related code

### PR DESCRIPTION
**Pull request by fzliu**
- Add a `BaseRepr` object from which all representation classes inherit
- Completed `DataframeRepr` based on example YAML, updated annotation-based initialization functions
- Updated dataframe-related code (still a WIP, needs further edits based on internal discussions)
- Updated docstrings and docstring formats for several DAG-related files
- Removed `DFIterator`, iterators can now be called directly from `Dataframe` objects
- Other slight formatting modifications

### CODE REVIEW

File: towhee/base_repr.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        Returns:
+            A string containing the serialized version of this representation. Example
+            output:
+
+            'VariableRepr:
+                name: variable
+                vtype: tensor
+                dtype: float'
+        """
+        out = self.__class__.__name___ + '/n'
```
Comment (junjiejiangjjj) [inline towhee/base_repr.py:45]: '/n' or '\n'

File: towhee/dag/graph_repr.py [hunk 2]
```diff
@@ -12,23 +12,37 @@
...
     """
 
     def __init__(self, builder_id: str):
         self.builder_id = builder_id
-        self._op_dict = {}
-        self._var_dict = None
+        self._op_dict = []
+        self._df_dict = []
 
     def op(self):
```
Comment (junjiejiangjjj) [inline towhee/dag/graph_repr.py:30]: named ops?

File: towhee/base_repr.py (hunk 1 above)
Comment (fzliu) [reply towhee/base_repr.py:45]: Typo, fixed.

File: towhee/dag/graph_repr.py (hunk 2 above)
Comment (fzliu) [reply towhee/dag/graph_repr.py:30]: Op names can be accessed via `op.name`, it also would be slightly easier to de-serialize from YAML with ops defined in a list.

**Review by GuoRentong — CHANGES_REQUESTED** [review]

File: towhee/dag/graph_repr.py [hunk 3]
```diff
@@ -12,23 +12,37 @@
...
     """
 
     def __init__(self, builder_id: str):
         self.builder_id = builder_id
         self._op_dict = {}
-        self._var_dict = None
+        self._df_dict = {}
 
-    def op(self):
+    def get_op(self, name: str) -> OperatorRepr:
```
[the remark this answers is not in the document]
Comment (fzliu) [reply towhee/dag/graph_repr.py]: Fixed.

File: towhee/dag/operator_repr.py [hunk 4]
```diff
@@ -13,34 +13,45 @@
...
     @property
-    def outputs(self):
-        return self._outputs
-    
+    def df_out(self):
+        return self._df_out
+
+    @df_in.setter
+    def df_in(self, value):
+        if not isinstance(value, DataframeRepr):
```
[the remark this answers is not in the document]
Comment (fzliu) [reply towhee/dag/operator_repr.py]: Added a type hint as a workaround; with a static type checker like `mypy`, this would be a better solution.

**Review by GuoRentong — CHANGES_REQUESTED** [review]
need a rebase before merge.
please edit the commit messages and remove the minor ones, such as 'fix typo', 'add import', etc.

**Pull request closed without merging**